src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -154,15 +119,15 @@ protected AbstractEntityAIBasic(@NotNull final J job)
                  * If yes, transition to NEEDS_ITEM.
                  * and wait for new items.
                  */
-          new AITarget(() -> !itemsCurrentlyNeeded.isEmpty(), this::waitForNeededItems),
+          new AITarget(() -> this.getOwnBuilding().areItemsNeeded(), this::waitForNeededItems),
                 /*
                  * Wait for different tools.
                  */
-          new AITarget(() -> this.needsShovel, this::waitForShovel),
-          new AITarget(() -> this.needsAxe, this::waitForAxe),
-          new AITarget(() -> this.needsHoe, this::waitForHoe),
-          new AITarget(() -> this.needsPickaxe, this::waitForPickaxe),
-          new AITarget(() -> this.needsWeapon, this::waitForWeapon),
+          new AITarget(() -> this.getOwnBuilding().needsShovel(), this::waitForShovel),
+          new AITarget(() -> this.getOwnBuilding().needsAxe(), this::waitForAxe),
+          new AITarget(() -> this.getOwnBuilding().needsHoe(), this::waitForHoe),
+          new AITarget(() -> this.getOwnBuilding().needsPickaxe(), this::waitForPickaxe),
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1612)